### PR TITLE
use python 3.9.18 instead of 3.9.15

### PR DESCRIPTION
**Pull request by severo**
To upgrade on Ubuntu and pyenv, I had to do the following:

```
sudo apt install zlib1g zlib1g-dev libssl-dev libbz2-dev libsqlite3-dev
libncurses-dev libffi-dev liblzma-dev libreadline-dev graphviz
libgraphviz-dev
pyenv update
pyenv install 3.9.18
```

and in every project folder

```
poetry env use 3.9.18
poetry update python
```

### CODE REVIEW

**Review by albertvillanova — APPROVED** [review]
This was in my todo list. Thanks.

**Comment by severo** [comment]
^ 26 minutes (and counting) for the worker unit tests...

**Comment by severo** [comment]
I'll stop it and re-run

**Review by albertvillanova — CHANGES_REQUESTED** [review]
I think this commit could be in a separate PR: 0f58abb96150e28f7dca12aa3a5d85e869535501

**Comment by severo** [comment]
OK. I'll just wait for #2089 to be merged, then rebase and redo the work here, so that we don't have to ignore the vulnerability

**Comment by albertvillanova** [comment]
> OK. I'll just wait for #2089 to be merged, then rebase and redo the work here, so that we don't have to ignore the vulnerability

Are we going to be merging PRs with red CI until that is fixed in `datasets`, released, and updated in datasets-server?

**Comment by severo** [comment]
https://github.com/huggingface/datasets-server/pull/2089 has no red CI. We are disabling ArrayXD support for now from datasets-server

**Review by albertvillanova — APPROVED** [review]
Thanks!

**Review by albertvillanova — COMMENTED** [review]
I think we should also update the `DEVELOPER_GUIDE`.

**Comment by severo** [comment]
Merging, so that we can create the following PRs on the right Python version, but not planning to deploy before Monday

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -34,8 +34,8 @@ If you use pyenv:
 
 ```bash
 cd libs/libcommon/
-pyenv install 3.9.15
-pyenv local 3.9.15
+pyenv install 3.9.18
+pyenv local 3.9.18
```
Comment (albertvillanova) [inline DEVELOPER_GUIDE.md:38]: Sorry! I just saw it now...

Comment (severo) [reply DEVELOPER_GUIDE.md:38]: Yes, I think the developer guide is OK. There is no section for "update to a new version of python" but I'm not sure we need it

**Pull request merged**